src/matching/matcher.ts

```diff
@@ -0,0 +1,78 @@
+import { Position } from './../motion/position';
+
+/**
+ * PairMatcher finds the position matching the given character, respecting nested
+ * instances of the pair.
+ */
+export class PairMatcher {
+  static pairings: { [key: string]: { match: string, nextMatchIsForward: boolean, limited?: boolean }} = {
+    "(" : { match: ")",  nextMatchIsForward: true  },
+    "{" : { match: "}",  nextMatchIsForward: true  },
+    "[" : { match: "]",  nextMatchIsForward: true  },
+    ")" : { match: "(",  nextMatchIsForward: false },
+    "}" : { match: "{",  nextMatchIsForward: false },
+    "]" : { match: "[",  nextMatchIsForward: false },
+    // These characters can't be used for "%"-based matching, but are still
+    // useful for text objects. TODO limited is a terrible name.
```

Comment:
Haha, nice, I was going to suggest you change limited to something else also. 

matchesWithPercentageMotion might be ok.